File: src/main/scala/com/github/unchama/buildassist/listener/BlockPlaceEventListener.java
package com.github.unchama.buildassist.listener;

import com.github.unchama.buildassist.BuildAssist;
import com.github.unchama.buildassist.PlayerData;
import com.github.unchama.buildassist.Util;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.block.BlockPlaceEvent;
import scala.collection.mutable.HashMap;

import java.math.BigDecimal;
import java.util.UUID;


public class BlockPlaceEventListener implements TypedEventListener<BlockPlaceEvent> {
    private final HashMap<UUID, PlayerData> playermap = BuildAssist.playermap();

    @Override
    @EventHandler
    public void onEvent(final BlockPlaceEvent event) {
        final Player player = event.getPlayer();
        if (!Util.inTrackedWorld(player)) {
            return;
        }

        final UUID uuid = player.getUniqueId();
        final PlayerData playerdata = playermap.getOrElse(uuid, () -> null);

        //プレイヤーデータが無い場合は処理終了
        if (playerdata == null) return;

        Util.addBuild1MinAmount(player, BigDecimal.ONE);
    }
}

File: src/main/scala/com/github/unchama/buildassist/listener/BlockPlaceEventListener.scala
package com.github.unchama.buildassist.listener

import java.math.BigDecimal

import com.github.unchama.buildassist.{BuildAssist, Util}
import com.github.unchama.seichiassist.ManagedWorld._
import org.bukkit.event.EventHandler
import org.bukkit.event.block.BlockPlaceEvent

/**
 * Created by karayuu on 2020/10/07
 */
class BlockPlaceEventListener extends TypedEventListener[BlockPlaceEvent] {
  @EventHandler
  override def onEvent(event: BlockPlaceEvent): Unit = {
    val player = event.getPlayer
    if (!player.getWorld.shouldTrackBuildBlock) return

    //設置がキャンセルされていたら終了
    if (event.isCancelled) return

    val uuid = player.getUniqueId
    val playerdata = BuildAssist.playermap.getOrElse(uuid, () => null)

    //プレイヤーデータが無い場合は処理終了
    if (playerdata == null) return

    Util.addBuild1MinAmount(player, BigDecimal.ONE)
  }
}
